src/main/java/org/embulk/parser/poi_excel/visitor/embulk/StringCellVisitor.java

```diff
@@ -31,7 +32,10 @@ protected String toString(Column column, Object source, double value) {
             }
         }
 
-        String s = Double.toString(value);
+        // Use toPlainString to avoid scientific notation
+        String s = BigDecimal.valueOf(value).toPlainString();
+        // Remove the trailing zeros of the zero scale value
+        // See: https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#toPlainString--
```

Comment:
Fetched from the following PRs.
https://github.com/trocco-io/embulk-parser-poi_excel/pull/5
https://github.com/trocco-io/embulk-parser-poi_excel/pull/6